oap-storage/src/main/java/oap/storage/MemoryStorage.java

```diff
@@ -46,10 +46,10 @@
 @Slf4j
 public class MemoryStorage<T> implements Storage<T>, ReplicationMaster<T> {
     public final Identifier<T> identifier;
-    protected final Lock lock;
+    public final Lock lock;
     private final List<DataListener<T>> dataListeners = new ArrayList<>();
     private final ArrayList<Constraint<T>> constraints = new ArrayList<>();
-    protected volatile ConcurrentMap<String, Metadata<T>> data = new ConcurrentHashMap<>();
+    public volatile ConcurrentMap<String, Metadata<T>> data = new ConcurrentHashMap<>();
```

Comment:
keep it package private and put mongo persistence into the same package of a different module.